superset/utils/cache.py

```diff
@@ -129,14 +122,11 @@ def wrapped_f(*args: Any, **kwargs: Any) -> Any:
             if not kwargs.get("cache", True):
                 return f(*args, **kwargs)
 
-            if key:
-                # format the key using args/kwargs passed to the decorated function
-                signature = inspect.signature(f)
-                bound_args = signature.bind(*args, **kwargs)
-                bound_args.apply_defaults()
-                cache_key = key.format(**bound_args.arguments)
-            else:
-                cache_key = view_cache_key(*args, **kwargs)
```

Comment:
`memoized_func()` has no case with `key = None` so unnecessary `view_cache_key()` can be removed.